Tests/scripts/hook_validations/structure.py

```diff
@@ -18,295 +29,281 @@
-        SCRIPT_YML_REGEX,
-        INCIDENT_FIELD_REGEX,
-        MISC_REGEX,
-        REPUTATION_REGEX
-    ]
-    SKIPPED_SCHEMAS = [
-        TEST_DATA_REGEX,
-        MISC_REGEX,
-        IMAGE_REGEX,
-        DESCRIPTION_REGEX,
-        PIPFILE_REGEX,
-        REPORT_REGEX,
-        SCRIPT_PY_REGEX,
-        SCRIPT_JS_REGEX,
-        SCRIPT_PS_REGEX,
-        INTEGRATION_JS_REGEX,
-        INTEGRATION_PY_REGEX,
-        INTEGRATION_PS_REGEX,
-        REPUTATION_REGEX,
-        BETA_INTEGRATION_YML_REGEX,
-        BETA_INTEGRATION_REGEX,
-        BETA_SCRIPT_REGEX,
-        BETA_PLAYBOOK_REGEX,
-    ]
-    REGEXES_TO_SCHEMA_DICT = {
-        INTEGRATION_REGEX: "integration",
-        INTEGRATION_YML_REGEX: "integration",
-        PLAYBOOK_REGEX: "playbook",
-        TEST_PLAYBOOK_REGEX: "test-playbook",
-        SCRIPT_REGEX: "script",
-        SCRIPT_YML_REGEX: "script",
-        WIDGETS_REGEX: "widget",
-        DASHBOARD_REGEX: "dashboard",
-        CONNECTIONS_REGEX: "canvas-context-connections",
-        CLASSIFIER_REGEX: "classifier",
-        LAYOUT_REGEX: "layout",
-        INCIDENT_FIELD_REGEX: "incidentfield",
+        Attributes:
+            file_path (str): the path to the file we are examining at the moment.
+            is_valid (bool): the attribute which saves the valid/in-valid status of the current file. will be bool only
+                             after running is_file_valid.
+            is_added_file (bool): whether the file is modified or added.
+        """
+    SCHEMAS_PATH = "Tests/schemas/"
+
+    FILE_SUFFIX_TO_LOAD_FUNCTION = {
+        '.yml': yaml.safe_load,
+        '.json': json.load,
+    }
+    SCHEMA_TO_REGEX = {
+        'integration': YML_INTEGRATION_REGEXES,
+        'playbook': YML_ALL_PLAYBOOKS_REGEX,
+        'script': YML_SCRIPT_REGEXES,
+        'widget': JSON_ALL_WIDGETS_REGEXES,
+        'dashboard': JSON_ALL_DASHBOARDS_REGEXES,
+        'canvas-context-connections': JSON_ALL_CONNECTIONS_REGEXES,
+        'classifier': JSON_ALL_CLASSIFIER_REGEXES,
+        'layout': JSON_ALL_LAYOUT_REGEXES,
+        'incidentfield': JSON_ALL_INCIDENT_FIELD_REGEXES,
     }
 
-    SCHEMAS_PATH = "Tests/schemas/"
+    PATHS_TO_VALIDATE = {
+        'python-script': PYTHON_ALL_REGEXES,
+        'reports': JSON_ALL_REPORTS_REGEXES,
+        'reputation': [MISC_REPUTATIONS_REGEX],
+        'reputations': [MISC_REGEX]
+    }
 
     def __init__(self, file_path, is_added_file=False, is_renamed=False):
-        self._is_valid = True
+        # type: (str, bool, bool) -> None
+        self.is_valid = None  # type: Optional[bool]
         self.file_path = file_path
         self.is_added_file = is_added_file
         self.is_renamed = is_renamed
+        self.scheme_name = self.scheme_of_file_by_path()
+        self.file_type = self.get_file_type()
+        self.current_file = self.load_data_from_file()
+        self.old_file = None if (is_renamed or is_added_file) else get_remote_file(self.file_path)
```

Comment:
if you rename, you should still get the original file (no reason to skip BC validations for renamed files)